lib/utils/FileCache.cjs

```diff
@@ -85,25 +86,6 @@ class FileCache {
 		return changed;
 	}
 
-	/**
-	 * @param {string} key
-	 * @param {string | undefined} fileHash
-	 * @returns {boolean}
-	 */
-	#hasFileCheckSumChanged(key, fileHash) {
-		if (!fileHash) {
-			return true;
-		}
-
-		const metaCache = this._fileCache.cache.getKey(key);
-
-		if (fileHash !== metaCache?.hash) {
-			return true;
-		}
```

Comment:
This condition would never match and this function would never return `true` because `fileHash` and `metaCache?.hash` are always the same value.